src/browser/BrowserAction.cpp

```diff
@@ -263,21 +264,31 @@ QJsonObject BrowserAction::handleGetLogins(const QJsonObject& json, const QStrin
     const QString formUrl = decrypted.value("submitUrl").toString();
     const QString auth = decrypted.value("httpAuth").toString();
     const bool httpAuth = auth.compare(TRUE_STR) == 0;
-    const QJsonArray users = browserService()->findMatchingEntries(id, siteUrl, formUrl, "", keyList, httpAuth);
+    auto requestId = decrypted.value("requestID").toString();
 
-    if (users.isEmpty()) {
-        return getErrorReply(action, ERROR_KEEPASS_NO_LOGINS_FOUND);
-    }
+    if (browserService()->isAccessConfirmRequested()) {
+        auto errorReply = getErrorReply(action, ERROR_KEEPASS_ACTION_CANCELLED_OR_DENIED);
 
-    const QString newNonce = browserMessageBuilder()->incrementNonce(nonce);
+        if (!requestId.isEmpty()) {
+            errorReply["requestID"] = requestId;
+        }
 
-    QJsonObject message = browserMessageBuilder()->buildMessage(newNonce);
-    message["count"] = users.count();
-    message["entries"] = users;
-    message["hash"] = hash;
-    message["id"] = id;
+        return errorReply;
+    }
 
-    return buildResponse(action, message, newNonce);
+    browserService()->findEntries(socket,
+                                  incrementedNonce,
+                                  m_clientPublicKey,
+                                  m_secretKey,
+                                  id,
+                                  hash,
+                                  requestId,
```

Comment:
Just a note: we might want to create a general "Request Struct" that has all this info baked into it and we can pass around to various functions that know what to do with the elements. Would reduce the variable jumble.